### PR TITLE
Version upgrade updates

### PR DESCRIPTION
**Pull request by janinefreeman**
Updates the version upgrade script to make the language clearer for the adjust variable changes (from table with semicolons to separate variables with underscores) and fixes the logic for upgrading the "enable custom time period" losses in that widget, which were not upgrading correctly in the current dev version.

Please specifically test enabling a custom time period loss in SAM 2022.11.21 and upgrading that file, and check that the loss upgrades correctly.

### CODE REVIEW

File: deploy/runtime/versions.lk [hunk 1]
```diff
@@ -81,86 +87,101 @@ version_upgrade{'2023.9.19'} = define() {
...
+		value('dc_adjust_en_timeindex', x.en_timeindex);
+		value('dc_adjust_timeindex', x.timeindex);
+		
+		//AC adjust factor
+		x = oldvalue('adjust');
+		message('AC losses variable format changed from single table ("adjust") to separate variables: adjust_constant (number), adjust_en_timeindex (number), adjust_timeindex (array), adjust_en_periods (number), adjust_periods (array).');
+		value('adjust_constant', x.constant);
+		value('adjust_en_periods', x.en_periods);
+		p = x.periods; //this logic works for both blank period arrays and for completed ones
+		value('adjust_periods', p);
```
Comment (mjprilliman) [inline deploy/runtime/versions.lk:110]: adjust_periods_dc fails on simulate when upgrading the default detailed PV case (without changing anything)

image

**Comment by mjprilliman** [comment]
Will close #1604

File: deploy/runtime/versions.lk (hunk 1 above)
Comment (janinefreeman) [reply deploy/runtime/versions.lk:110]: I can't replicate this on my machine. However, what is getting loaded in seems to have changed and is no longer working for me. Possibly a wxwidgets issue? Let's talk more at our working session...

**Comment by mjprilliman** [comment]
[version-upgrade-testing-csp.zip](https://github.com/NREL/SAM/files/13506311/version-upgrade-testing-csp.zip)
My goal is to run the cases in these two files through the upgrade script and have them simulate without any error.

**Review by mjprilliman — APPROVED** [review]
The last commit fixed the issues I was finding on the first review. Thanks for working through this!

**Pull request merged**